rust/main/chains/hyperlane-starknet/src/indexer.rs

```diff
@@ -262,6 +262,17 @@ async fn fetch_logs_in_range<T>(
 where
     T: std::fmt::Debug,
 {
+    // sanity check the range, because the provider doesn't return an error if the range is invalid
```

Comment:
So I understand correctly, this won't fully protect us but adds another layer of defense? as we already do a "get the latest block with one RPC, then try to index up to that latest block in the next RPC" and the issue is that the provider will load balance to different underlying nodes for the two RPCs, and we can end up in the same situation here where this get block height RPC is still load balanced to the node that's ahead but the getLogs RPC is still sent to a node that's behind? But the idea is this can be less likely to happen by doing the block height RPCs twice

Reply:
sg, happy to merge, it will increase our RPCs made but that's probably fine